core/app/models/spree/order.rb

```diff
@@ -329,6 +329,10 @@ def create_tax_charge!
     def outstanding_balance
       if state == 'canceled'
         -1 * payment_total
+      elsif reimbursements.includes(:refunds).size > 0
```

Comment:
Shouldn't we take refunds into account for the canceled state above as well? I.e. if you got a refund and then canceled your order, we should subtract the amount of reimbursements from the amount we now owe you, right? I'm thinking something like this:
```ruby
      # If reimbursement has happened add it back to total to prevent balance_due payment state
      # See: https://github.com/spree/spree/issues/6229
      adjusted_payment_total = payment_total + refund_total
      if state == 'canceled'
        -1 * adjusted_payment_total
      else
        total - adjusted_payment_total
      end
```

Simplifies the code a bit, too?